stm32_base/hal/usart_stm32/usart_device_stm32.c

```diff
@@ -648,6 +648,22 @@ int usart3_readLine(char *s, int len)
 // Interrupt Handlers
 ////////////////////////////////////////////////////////////////////////////
 
+inline static void handle_usart_overrun(USART_TypeDef* USARTx)
+{
+	uint32_t cChar;
+    if (USART_GetITStatus(USARTx, USART_IT_ORE_RX) != RESET) {
```

Comment:
So this is a good example of where I would use a return style if check here to avoid indentation.  I'm not saying do this, but consider it.  I might do something like

```C
 if (USART_GetITStatus(USARTx, USART_IT_ORE_RX) != SET)
   return;

...
```

See how that could save you lots of indentation.  This way also works.